File: quivi.backoffice.react.webapp/src/hooks/api/Dtos/reporting/GetPartnerChargeMethodSalesRequest.ts
import { ChargePartner } from "../acquirerconfigurations/ChargePartner";
import { ChargeMethod } from "../ChargeMethod";
import { PagedRequest } from "../PagedRequest";
import { SalesPeriod } from "./SalesPeriod";

export interface GetPartnerChargeMethodSalesRequest extends PagedRequest {
    readonly adminView?: boolean;
    readonly chargePartners?: ChargePartner[];
    readonly chargeMethods?: ChargeMethod[];
    readonly period?: SalesPeriod;
    readonly from?: string;
    readonly to?: string;
}

File: Quivi.Backoffice.Api/Requests/Reporting/GetPartnerChargeMethodSalesRequest.cs
using Quivi.Domain.Entities.Charges;
using Quivi.Infrastructure.Abstractions.Repositories.Criterias;

namespace Quivi.Backoffice.Api.Requests.Reporting
{
    public class GetPartnerChargeMethodSalesRequest : APagedRequest
    {
        public bool? AdminView { get; init; }
        public IEnumerable<ChargeMethod>? ChargeMethods { get; init; }
        public IEnumerable<ChargePartner>? ChargePartners { get; init; }
        public SalesPeriod? Period { get; init; }
        public ProductSalesSortBy SortBy { get; init; }
        public DateTimeOffset? From { get; init; }
        public DateTimeOffset? To { get; init; }
    }
}